api/buffer/design.go

```diff
@@ -343,6 +343,7 @@ var Receiver = Type("Receiver", func() {
 	receiverFields()
 	Attribute("exports", ArrayOf(Export), func() {
 		Description("List of exports, max 20 exports per a receiver.")
+		Example([]any{ExampleExport()})
```

Comment:
Ono sa to interne snazilo spojit vygenerovany example `map[string]any` a tvoj example `ExampleExportDef` a spravit z toho `Array`. Ked som definoval example rucne tak to preslo.

Mozno by stalo za to [zalozit issue:](https://github.com/goadesign/goa/issues)
- mam nejaky `Custom` type s definovaným `Example`, ktory je struct.
- mam niekde inde `ArrayOf(Custom)`, bez definovaneho `Example`.
- nastane `panic`
- mohol by sa tam linknut tento commit s fixom

--- 

Ak je v hlaske `ExampleExportDef`, tak som nasiel, kde vsade je `ArrayOf(Export)` a fixol som to.

Reply:
Skoda, ze je tam viacero takych veci v tom `Goa` frameworku, su to vacsino nejake drobnosti, ktore by nemal byt problem fixnut.

Reply:
Hm, já si nejsem jistej že chápu v čem je problém a proč to ten commit fixnul. 😕

Reply:
Ok, tak skusim este raz:
- Mame tu typ `ArrayOf(Export)/ArrayOf(Receiver)/....`, 
- Ak pre nejaky typ nie je definovany `Example(...)`, tak ho lib-ka generuje.
- Nema(lo) to definovany `Example(...)`, tak sa generoval, je to tato cast zo stacktrace:
  - https://github.com/goadesign/goa/blob/c7deb3b6cc6741e4bc09965753d398ca1d9dd411/expr/example.go#L74-L76
-  Konkretne to spadne pocas generovania example TU, v metode `MakeSlice`:
  - https://github.com/goadesign/goa/blob/c7deb3b6cc6741e4bc09965753d398ca1d9dd411/expr/types.go#L430-L436
  - 
![image](https://user-images.githubusercontent.com/19371734/211595614-07fa8f9e-f937-48f7-9807-f6ba65b6cffc.png)
- A tam je nejaky bug v Goa:
  -  :red_circle: pre vsetky `ArrayOf(<object>)` to ocakava example ako `[]map`, tak to urci `toReflectType`
  - :orange_circle:  no nas example ma typ `ExampleReceiverDer`.
- Ale na inych miestach to funguje spravne, problem/bug to je len s `ArrayOf`.
- A to, ze sa pre `ArrayOf` zada **example manualne je workaround**, ... generovanie example sa nespusti a nespadne to.
  
  
--- 

Je to takto jasnejsie?

Reply:
Ahá, tohle mi nějak uniklo:
> Ak pre nejaky typ nie je definovany Example(...), tak ho lib-ka generuje.
Už to začíná dávat smysl, díky!